### PR TITLE
Avoid entry_points deprecation warning

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/Lightning-AI/lightning/issues/13989

### Does your PR introduce any breaking changes? If yes, please list them.

None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [n/a] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [x] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

cc @borda @justusschock @kaushikb11 @awaelchli @ninginthecloud @akihironitta

### CODE REVIEW

File: src/pytorch_lightning/CHANGELOG.md [hunk 1]
```diff
@@ -49,6 +49,9 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - Fixed the `NeptuneLogger` dependency being unrecognized ([#13988](https://github.com/Lightning-AI/lightning/pull/13988))
 
 
+- Avoid `metadata.entry_points` depreaction warning on Python 3.10 ([#14052](https://github.com/Lightning-AI/lightning/pull/14052))
```
Comment (adamjstewart) [inline src/pytorch_lightning/CHANGELOG.md:52]:
```suggestion
- Avoid `metadata.entry_points` deprecation warning on Python 3.10 ([#14052](https://github.com/Lightning-AI/lightning/pull/14052))
```

**Review by awaelchli — APPROVED** [review]

**Comment by adamjstewart** [comment]
I can confirm that this fixes #13989 for me!

**Review by akihironitta — APPROVED** [review]
LGTM!

As #9980 has been merged, It's now running with Python 3.10 in "full" CPU testing CI. cc @carmocca @adamjstewart

Do you think we should let pytest treat `DeprecationWarning` as errors just like `FutureWarning`?
https://github.com/Lightning-AI/lightning/blob/d29a552b3c701ebc14d608347c1dbf55c3dfaa6a/setup.cfg#L35-L38

**Comment by carmocca** [comment]
> Do you think we should let pytest treat `DeprecationWarning` as errors just like `FutureWarning`?

I tried it in the past and there's a TON of overrides we would need to add, because we also would fail from warnings triggered inside our dependencies. Unless there's an easy way to filter those triggered in our codebase.

**Review by Borda — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Comment by adamjstewart** [comment]
> Do you think we should let pytest treat `DeprecationWarning` as errors just like `FutureWarning`?

We do this in [TorchGeo](https://github.com/microsoft/torchgeo/blob/main/pyproject.toml#L71,L75), including some specific to lightning.

> there's a TON of overrides we would need to add, because we also would fail from warnings triggered inside our dependencies. Unless there's an easy way to filter those triggered in our codebase.

This is unfortunately true in my experience, although you probably wouldn't need more than 5–10. I'm unaware of any way to handle deprecation warnings differently depending on whether your code or a dependency is using a deprecated feature.

**Pull request merged**